packages/kbn-management/settings/components/form/form.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { Fragment } from 'react';
+
+import type { FieldDefinition } from '@kbn/management-settings-types';
+import { FieldRow, OnChangeFn } from '@kbn/management-settings-components-field-row';
+import { SettingType, UnsavedFieldChange } from '@kbn/management-settings-types';
+import { isEmpty } from 'lodash';
+import { BottomBar } from './bottom_bar';
+
+export interface FormProps {
+  fields: Array<FieldDefinition<SettingType>>;
+  save: (changes: Record<string, UnsavedFieldChange<SettingType>>) => void;
```

Comment:
nit: we could consider creating a service function-- in `services.tsx` called `saveAll` or similar-- which we could use in this form to save the changes.  It just depends on if we feel the `application` should save the changes, or if the form could do it with a service function.  I'm leaning towards providing a service function, but I'm curious what you think... (we have a lot of drilled props at the moment, so it would be great to actually _remove_ one).

Reply:
Yes, it makes sense to add the saving function in `services.tsx` as I see now that this is the approach that is used in the field input/row packages (e.g. the `showDanger` function in field input) and, as you mentioned, it would be good to remove this drilled prop.

Added this change with https://github.com/elastic/kibana/pull/166460/commits/ad18ae37177353621d2facc1ab96c766bb9dc8f6